src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -372,6 +459,46 @@ int TieredHNSWIndex<DataType, DistType>::addVector(const void *blob, labelType l
     return 1;
 }
 
+template <typename DataType, typename DistType>
+int TieredHNSWIndex<DataType, DistType>::deleteVector(labelType label) {
+    int num_deleted_vectors = 0;
+    this->flatIndexGuard.lock_shared();
+    if (this->flatBuffer->isLabelExists(label)) {
+        this->flatIndexGuard.unlock_shared();
+        this->flatIndexGuard.lock();
+        // Check again if the label exists, as it may have been removed while we released the lock.
+        if (this->flatBuffer->isLabelExists(label)) {
+            // Invalidate the pending insert job(s) into HNSW associated with this label
+            for (auto *job : this->labelToInsertJobs.at(label)) {
+                reinterpret_cast<HNSWInsertJob *>(job)->id = INVALID_JOB_ID;
+            }
+            // Remove the pending insert job(s) from the labelToInsertJobs mapping.
+            this->labelToInsertJobs.erase(label);
+            auto ids = this->flatBuffer->getIdsOfLabel(label);
+            // Go over the every id that corresponds the label and remove it from the flat buffer.
+            // We fetch the ids every time since in case of multi value, we delete ids from the
+            // flat, and it triggers swap between the last id and the deleted one.
```

Comment:
I don't really see the point since in most cases it won't save us swapping, as the specific ids of the label are not necessarily the last ones (if there are more labels stored in the flat index)